bundles/org.openhab.binding.somneo/src/main/java/org/openhab/binding/somneo/internal/SomneoHandler.java

```diff
@@ -0,0 +1,542 @@
+            }
+        } catch (TimeoutException e) {
+            logger.warn("Timeout: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (Exception e) {
+            if (e.getCause() instanceof EOFException) {
+                // Occurs on parallel mobile app access
+                logger.warn("EOF: {}", e.getMessage());
+            } else {
+                logger.error("Sending command failed", e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        initConnector();
+        updateThingProperties();
+        startPolling();
+    }
+
+    @Override
+    public void dispose() {
+        stopPolling();
+        stopRemainingTimer();
+
+        super.dispose();
+    }
+
+    private void initConnector() {
+        if (connector == null) {
+            SomneoConfiguration config = getConfigAs(SomneoConfiguration.class);
+            HttpClient httpClient;
+            if (config.ignoreSSLErrors) {
+                logger.info("Using the insecure client for thing '{}'.", thing.getUID());
+                httpClient = httpClientProvider.getInsecureClient();
+            } else {
+                logger.info("Using the secure client for thing '{}'.", thing.getUID());
+                httpClient = httpClientProvider.getSecureClient();
+            }
+
+            connector = new SomneoHttpConnector(config, httpClient);
+        }
+    }
+
+    private void updateThingProperties() {
+        final SomneoHttpConnector connector = this.connector;
+        if (connector == null) {
+            return;
+        }
+
+        Map<String, String> properties = editProperties();
+        properties.put(Thing.PROPERTY_VENDOR, PROPERTY_VENDOR_NAME);
+
+        try {
+            final DeviceData deviceData = connector.fetchDeviceData();
+            String value = deviceData.getModelId();
+            if (value != null) {
+                properties.put(Thing.PROPERTY_MODEL_ID, value);
+            }
+            value = deviceData.getSerial();
+            if (value != null) {
+                properties.put(Thing.PROPERTY_SERIAL_NUMBER, value);
+            }
+
+            final WifiData wifiData = connector.fetchWifiData();
+            value = wifiData.getMacAddress();
+            if (value != null) {
+                properties.put(Thing.PROPERTY_MAC_ADDRESS, value);
+            }
+
+            final FirmwareData firmwareData = connector.fetchFirmwareData();
+            value = firmwareData.getVersion();
+            if (value != null) {
+                properties.put(Thing.PROPERTY_FIRMWARE_VERSION, value);
+            }
+
+            updateProperties(properties);
+        } catch (Exception e) {
+            logger.error("Fetching data failed", e);
```

Comment:
Log level should be reduced to **debug**, but probably it would be better to skip this logging as all information is given below when updating status, which is also logged.